src/transport/FabricTable.h

```diff
@@ -324,20 +326,19 @@ class ConstFabricIterator
 class DLL_EXPORT FabricTable
 {
 public:
+    FabricTable() { Reset(); }
     CHIP_ERROR Store(FabricIndex id);
     CHIP_ERROR LoadFromStorage(FabricIndex id);
-    CHIP_ERROR Delete(FabricIndex id);
 
-    FabricInfo * AssignFabricIndex(FabricIndex fabricId);
+    CHIP_ERROR Delete(FabricIndex id);
+    void DeleteAllFabrics();
 
-    FabricInfo * AssignFabricIndex(FabricIndex fabricId, NodeId nodeId);
+    CHIP_ERROR AddNewFabric(FabricInfo & fabric, FabricIndex & assignedIndex);
```

Comment:
Please make assignedIndex a `FabricIndex *` so it's clearer at callsites that it's an outparam.

And please add documentation explaining the semantics here.